Vonk.Plugin.DocumentOperation/DocumentService.cs

```diff
@@ -84,35 +84,39 @@ public async Task Document(IVonkContext vonkContext, string compositionID)
             var composedBundle = CreateEmptyBundle(operationRequestPath);
 
             // Get Composition resource
-            (bool allResourceResolved, Resource resolvedResource, string failedReference) = await ResolveResource(compositionID, "Composition");
-
-            if (allResourceResolved)
+            (bool compositionResolved, Resource resolvedResource, string failedReference) = await ResolveResource(compositionID, "Composition");
+            if (compositionResolved)
             {
                 // Include Composition resource in search results
                 composedBundle.Total = 1;
                 composedBundle.AddSearchEntry(resolvedResource, "Composition/" + compositionID, Bundle.SearchEntryMode.Match);
 
                 // Recursively resolve and include all references in the search bundle
-                (allResourceResolved, failedReference) = await IncludeReferencesInBundle(resolvedResource, composedBundle);
-            }
-            else  // Composition resource, on which the operation is called, does not exist
-            {
-                _logger.LogTrace("$document called on non-existing Composition/{id}", compositionID);
-                composedBundle.Total = 0;
+                bool includedResourcesResolved;
+                (includedResourcesResolved, failedReference) = await IncludeReferencesInBundle(resolvedResource, composedBundle);
             }
 
             // Handle responses
             IVonkResponse response = vonkContext.Response;
             vonkContext.Arguments.Handled(); // Signal to Vonk -> Mark arguments as "done"
-            if (!allResourceResolved)
+            if (!failedReference.Equals(string.Empty))
             {
-                CancelDocumentOperation(response, LocalReferenceNotResolvedIssue(failedReference));
+                if (!compositionResolved) // Composition resource, on which the operation is called, does not exist
+                {
+                    _logger.LogTrace("$document called on non-existing Composition/{id}", compositionID);
+                    composedBundle.Total = 0;
+                    CancelDocumentOperation(response, LocalReferenceNotResolvedIssue(failedReference, IssueSeverity.Warning), StatusCodes.Status200OK, composedBundle);
+                }
+                else
+                {
+                    CancelDocumentOperation(response, LocalReferenceNotResolvedIssue(failedReference));
```

Comment:
So if the Composition itself is missing, it returns 200, but if any of the references in the Composition is missing, it returns 500. That looks inconsistent to me.